packages/backend-storage/src/construct.ts

```diff
@@ -146,15 +151,22 @@ export class AmplifyStorage
    * Store storage outputs using provided strategy
    */
   private storeOutput = (
-    outputStorageStrategy: BackendOutputStorageStrategy<StorageOutput> = new StackMetadataBackendOutputStorageStrategy(
-      Stack.of(this)
-    )
+    outputStorageStrategy: BackendOutputStorageStrategy<{
+      version: '1';
+      payload: Record<string, string>;
+    }> = new StackMetadataBackendOutputStorageStrategy(Stack.of(this)),
+    isDefault: boolean = false
   ): void => {
+    /* The following code can guarantee there's only one `isDefault`
+     * because if we can only create one construct with the same `storageRegion` and `bucketName` name.
+     * The default bucket takes the `storageRegion` and `bucketName` name without a number post-fix.
+     */
+    const num = isDefault ? '' : Math.floor(Math.random() * 100);
```

Comment:
We shouldn't use random generator here. This will cause re-deployments of storage as outputs are going to change every time impacting performance (e.g. this will cause sandbox never hotswapping).
Instead of this we should either apply consistent hashing OR have global counter for storage instances.

Reply:
Thanks. Yea, I'm looking for some suggestions to replace the random generator. I'll try those.

Reply:
We should do the same thing that we do for auth and data: https://github.com/aws-amplify/amplify-backend/blob/7361ff4ed9da10f9b48e327c96eff25eb81a9e10/packages/backend-auth/src/factory.ts#L68

keep a static counter in the factory class

Reply:
Done cddcd97830e605666c3d3c8be81e4f2fedbdb5b5